macros/contexts/contextBoolean.pl

```diff
@@ -0,0 +1,253 @@
+			TeX           => '\vee ',
+			perl          => '||',
+			alteratives   => ['\x{2228}'],
+		},
+		'and' => {
+			class         => 'context::Boolean::BOP::and',
+			precedence    => $andPrec,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' and ',
+			TeX           => '\wedge ',
+			perl          => '&&',
+			alteratives   => ['\x{2229}'],
+		},
+		'xor' => {
+			class         => 'context::Boolean::BOP::xor',
+			precedence    => $andPrec,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' xor ',
+			perl          => '!=',
+			TeX           => '\oplus ',
+			alteratives   => [ '\x{22BB}', '\x{2295}' ],
+		},
+		# unary negation is only allowed if the '-' token is recognized, so make it BOP::undef
+		'-' => {
+			class         => 'Parser::BOP::undefined',
+			precedence    => $notPrec - 1,
+			associativity => 'left',
+			type          => 'both',
+			string        => '-',
+		},
+		'not' => {
+			class            => 'context::Boolean::UOP::not',
+			precedence       => $notPrec,
+			associativity    => 'left',
+			type             => 'unary',
+			string           => ' not ',
+			TeX              => '\mathord{\sim}',
+			perl             => '!',
+			alteratives      => [ '\x{172}', '\x{732}' ],
+			nofractionparens => 1,
+		},
+		' '  => { alias => 'and' },
+		'*'  => { alias => 'and' },
+		'+'  => { alias => 'or' },
+		'!'  => { alias => 'not' },
+		'u-' => { alias => 'not' },
+		'><' => { alias => 'xor' },
+	);
+
+	## Set default variables 'p' and 'q'
+	$context->variables->are(
+		p => { isValue => 1, type => Value::Type('Boolean') },
+		q => { isValue => 1, type => Value::Type('Boolean') },
+	);
+}
+
+## Subclass Value::Formula for boolean formulas
+package context::Boolean::Formula;
+our @ISA = ('Value::Formula');
+
+sub new {
+	my $class = shift;
+	my $self  = $class->SUPER::new(@_);
+
+	my @variables = $self->{context}->variables->names;
+	my @points;
+	my @values;
+
+	my $f = $self->{f} = $self->perlFunction(undef, \@variables);
+
+	foreach my $combination (0 .. 2**@variables - 1) {
+		my @point = map { $combination & 2**$_ ? 1 : 0 } (0 .. $#variables);
+		my $value = &$f(@point) ? 1 : 0;
+		push @points, \@point;
+		push @values, $value;
+	}
+
+	$self->{test_points} = \@points;
+	$self->{test_values} = \@values;
+
+	return $self;
+}
+
+sub typeRef { return Value::Type('Boolean') }
+
+package context::Boolean::BOP;
+our @ISA = qw(Parser::BOP);
+
+sub _check {
+	my $self = shift;
+	# constants don't have type methods
+	# TODO: should that be "fixed"?
+	my $lType = $self->{lop}->type || $self->{lop}{type};
+	my $rType = $self->{rop}->type || $self->{lop}{type};
+
+	$self->{type} = Value::Type('Boolean');
+}
+
+sub perl {
+	my $self   = shift;
+	my $result = $self->SUPER::perl(@_);
+	my $pkg    = $self->Package('Real');
+	return "$pkg->new($result ? 1 : 0)";
+}
+
+package context::Boolean::BOP::or;
+our @ISA = qw(context::Boolean::BOP);
+
+sub _eval { return $_[1] || $_[2] ? 1 : 0 }
```

Comment:
Yep, I figured this one out 👍

Reply:
Although it is also necessary to ensure that the result is a MathObject here as well, since returning a raw perl value will not allow ops higher in the tree to call `->value`...

So I have 
```perl
sub _eval { return $_[0]->Package('Boolean')->new($_[1]->value || $_[2]->value ? 1 : 0) }
```

Reply:
It is probably better to use

``` perl
sub _eval { return $_[1]->value || $_[2]->value ? $context::Boolean::T : $context::Boolean::F) }
```

so you don't have to keep creating new objects (the constructors do a lot of argument checking).  Otherwise, when you know the arguments are valid, use `make` rather than `new`, for efficiency.

Reply:
Makes sense to me 🤡